Readme (#6)
khealth/dialogs: readme.MD

# Dialogs Framework

> The Dialogs Framework is a Python library for writing resumable functions with persistent execution state.

Reviewer: ```suggestion
A Python library for creating resumable dialog interfaces
```

Author: Accepted but rephrased, I prefer not to use "dialog" in the definition

## Resumable what now?

> This framework is intended for long-lived backend flows, such as chatbot conversations. It allows one to write code as a chronological flow, and provides the magic to make execution resume where it left off.

Reviewer: Give an explanation for what flow is. I would also explain how it is being used in K.

Author: I added a paragraph but I couldn't write something clear enough about the K usage, I think it still sounds mysterious to non-users

> ## Code sample

Reviewer: ```suggestion
## Example
```

```python
# No dialog, manual state manipulation
def ask_a_name(client_answer, state):
    if not state.did_ask_already:
        state.did_ask_already = True
        name = client_answer
        return f"{name} is a beautiful name!"
    else:

>         return "What is your name?"

Reviewer: In a later section you can show how would it work without the framework and explain the drawbacks

Author: I want to keep them close together, I think seeing just the framework code isn't clear enough to the uninitiated.

# With the dialogs framework
@dialog()
def ask_a_name_dialog():
    run(send_message("What is your name?"))
    name = run(get_client_response())

>     run(send_message(f"{name} is a beautiful name!"))

Reviewer: Show the usage of the framework first.

Author: Switched 👍

```

## Philosophy
* No direct access to flow state.
* No breaks in the flow with `return` statements.

> * No in-memory state, scale-out friendly.

Reviewer: You should explain what flow is

Author: Removed it from the first point, not sure how to simplify the second
